src/explore-education-statistics-common/src/modules/table-tool/contexts/TableHeadersContext.tsx

```diff
@@ -78,6 +83,14 @@ export const TableHeadersContextProvider = ({
           : [...current, groupId],
       );
 
+    const toggleReverseOrder = (groupId: string) => {
```

Comment:
I don't really get why we need to store a list of the groups that have been reversed, especially as `reverseOrder` doesn't seem to be used anywhere.

I think that instead you can make sure that the read only list component updates when the values change by changing it to use `watch` instead of `getValues` (see https://www.react-hook-form.com/api/useform/watch/)